src/format_config.mli

```diff
@@ -0,0 +1,32 @@
+(** Represent the [(formatting)] field in [dune-project] files *)
+
+open! Stdune
+open Import
+
+(** Dune can format either source files through external programs (ocaml and
+    reason are builtin dialects) or dune files *)
+type language = Dialect of string | Dune
```

Comment:
Is it possible to make this `Dialect of Dialect.t`? Or does this introduce a circularity perhaps?

Reply:
I tried to do this, but unfortunately this would require access to the dialect DB during parsing.